### PR TITLE
Towards one-click release

### PR DESCRIPTION
**Pull request by kunaltyagi**
[CI](https://kunaltyagi.visualstudio.com/pcl/_build/results?buildId=394&view=results)
[Created Release](https://github.com/kunaltyagi/pcl/releases/tag/pcl-1.11.1-rc0)

Pre-release:
1. Bump version: using `.dev/scripts/bump_release.bash <version>`
2. Modify `CHANGES.md` (from documentation job or a release candidate)
3. Create PR, merge it

Release:
1. Start CI on master
2. Add variable:
  * `VERSION`: self-explanatory
  * `RC`: number with the pre-release wanted, 0 for final release
3. Run CI
4. Goto the releases, **edit tag** and Changelog, then finalize the release

Post Release: (Can be done with a CI, but would need to push someplace)
1. Bump version using `.dev/scripts/bump_post_release.bash`
2. Create PR, merge

---
~Missing:~
* ~Git tag~
* ~Git commit for pre and post release~

~@SergioRAgostinho @taketwo Shall we use the same SSH key (taketwo's I gather) used for documentation job for pushing updates?~

TODO in this PR:
* ~Make changes to CHANGES.md~
* ~Use git tag instead of user provided tag~
* Limit to 
  * ~Tagged ~commits on master, dividing release PR in 2 parts:
    * Pre-release: bump version in CMake and README, tag it, push it, PR it
    * Release: test, prepare, release, bump version

### CODE REVIEW

**Comment by SergioRAgostinho** [comment]
> @SergioRAgostinho @taketwo Shall we use the same SSH key (taketwo's I gather) used for documentation job for pushing updates?

If he's ok with it, I would say so. It's already installed. I think it would only make sense to use another ssh key if it was associated with a new email like releases@pointclouds.org. Something more on the organization side and less personal.

**Comment by taketwo** [comment]
Well, that's not my personal key. I just generated a pair, uploaded the private one to Azure, public to GitHub, and that was about the last time I saw either.

**Comment by kunaltyagi** [comment]
I'll add the `git tag` related steps in this CI using that key.

But the second part of my description needs updating and some thought.
> Tagged commits on master, dividing release PR in 2 parts:
>  *  Pre-release: bump version in CMake and README, tag it, push it, PR it
>  *  Release: test, prepare, release, bump version

It's not possible in pre-release to tag before merging to master. Neither is it possible to create a PR. IMO, pre-release can be made a script, but not a CI

Release can be triggered manually and it can tag the HEAD@master making the release pipeline:
* Manually run `start-release.bash`. It'll create a diff and commit it. Manually push, create PR
* Once merged to master, manually run the CI. The release CI tests, tags, prepares, releases.
* Manually run `post-release.bash`. It'll create a diff and commit it. Push it and create PR
* Merge the version bump

Not quite 1 push release. Thoughts welcome

**Comment by SergioRAgostinho** [comment]
I don't think we need to be that pedantic about it. I'm perfectly fine with a four-click release.

**Comment by kunaltyagi** [comment]
https://kunaltyagi.visualstudio.com/pcl/_build/results?buildId=394&view=results

Pre-release:
1. Bump version: using `.dev/scripts/bump_release.bash <version>`
2. Modify `CHANGES.md` (from documentation job or a release candidate)
3. Create PR, merge it

Release:
1. Start CI on master
2. Add variable:
  * `version`: self-explanatory
  * `isPreRelease`: number with the pre-release wanted, 0 for final release
3. Run CI
4. Goto the releases, **edit tag** and Changelog (if needed), then finalize the release

Post Release: (Can be done with a CI, but would need to push someplace)
1. Bump version using `.dev/scripts/bump_post_release.bash`
2. Create PR, merge

**Comment by kunaltyagi** [comment]
Should the changelog contain the changelog (current) or refer to look at CHANGES.md (like the recent releases)?

**Comment by SergioRAgostinho** [comment]
I propose to be similar to what you did for 1.11.0. It should contain the 2 or 3 lines summarizing the most important aspects and then refer to changes.md

**Comment by kunaltyagi** [comment]
PTAL

File: .ci/azure-pipelines/release.yaml [hunk 1]
```diff
@@ -93,40 +116,96 @@ stages:
...
-      # find the commit hash on a quick non-forced update too
-      fetchDepth: 10
+    - checkout: none
     - script: |
-        $(Build.SourcesDirectory)/.dev/scripts/generate_changelog.py --with-misc > $(CHANGELOG)
-        pandoc -f markdown -t plain --wrap=none $(CHANGELOG)
+        cat > $(CHANGELOG) <<EOF
+        <Insert Summary Here>
+
+        For an exhustive list of newly added features, deprecations and other changes
```
Comment (SergioRAgostinho) [inline .ci/azure-pipelines/release.yaml:164]: Typo in exhaustive.

Comment (kunaltyagi) [reply .ci/azure-pipelines/release.yaml:164]: exhausting :P

**Comment by kunaltyagi** [comment]
One image not showing, but: https://github.com/PointCloudLibrary/pcl/wiki/Preparing-Releases

**Pull request merged**